src/main/java/Lotto.java

```diff
@@ -17,29 +17,27 @@ public Lotto(int... nums) {
     }
 
     public Lotto(List<Integer> nums){
-        if(nums.stream().anyMatch(num -> num < Constants.MIN_LOTTO_NUMBER || num > Constants.MAX_LOTTO_NUMBER)){
-            throw new IllegalArgumentException("로또 번호는 1~46 범위이어야 합니다.");
-        }
-
-        this.nums = nums;
+        this.lottoNums = nums.stream()
+                .map(LottoNumber::new)
+                .collect(Collectors.toCollection(ArrayList::new));
     }
 
```

Comment:
```suggestion
    public Lotto(List<LottoNumber> nums){
```

주 생성자는 위와 같이 객체가 가지는 인스턴스 변수와 같은 타입이다.
주 생성자를 추가하고 부 생성자가 주 생성자를 호출하도록 구현해 본다.